App navigation flex

skjnldsv opened the pull request:
Flex on app-navigation to finish the reorganisation of the app-navigation done on february by me.
Still some old code to clean.

skjnldsv committed:
Simplification and reorganisation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Removed unused old vars and reorganize

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Icon auto-padding fix

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Allow counter inside a to be clickable

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Navigation flex magic

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Second menu padding and ellipsis fix

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix counter inside link and highlight submenu as well

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix app navigation positionning

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fixed collapsible and Folders

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Animate hide/show collapsible menus

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

nickvergessen commented:
I guess this will break a lot of apps again...

skjnldsv commented:
@nickvergessen I will try my best to help reestablishing everything again. But would you prefer we do nothing and leave this mess around? 😞

nickvergessen commented:
No, of course progress is good. But maybe you can post in the forum in the dev section that there changes about this incoming, so people can check their apps and update them when necessary ;)

skjnldsv commented:
I will check other apps myself.
I already have @ChristophWurst's request to do the mail menu, I will pr all the needed others :)

skjnldsv committed:
Organize app-navigation menu types

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

MorrisJobke commented:
One thing that annoys me from time to time, is that the app content container is the actual scrolling part and the navigation around it is fixed. That also means, that mobile browsers don't properly detect if you scroll in there as a "page scroll" and don't hide the header of the browser. That means that it is unusable screen space. Does this PR here addresses this or does it only change stuff in the header/menu itself?

skjnldsv commented:
@MorrisJobke not in this pr. But it's indeed an issue I'd like very much to investigate. Is there an opened one already?

skjnldsv committed:
Fix typo

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Removed animation breaking things and open submenu which arent .collapsible by default

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix editable entry design and flex

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Delete and edit animation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
merge animation into same prop

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Cleanup

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Add inline style padding

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Finally force the menu to be a first-child of app-navigation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Cleanup 2

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix menu submenu after eec58ae32b0adf35af3aa550c1b1fe21bc2e963a

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Syntax fix and collapse button padding if no icon on link

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
For testing, here is a test app which show all of the possibilities for the app-navigation.
[templatetest.tar.gz](https://github.com/nextcloud/server/files/1293699/templatetest.tar.gz)

skjnldsv committed:
Icon fix for sub-items

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@nextcloud/designers Please review!
I introduced some new stuff, based on the current guidelines.
Please take a look at the template test app which contains all possibilities and combination for the menu.
Tell me what you think!

[templatetest.tar.gz](https://github.com/nextcloud/server/files/1294676/templatetest.tar.gz)

skjnldsv commented:
We should probably add a feedback on the submenus of a deletable entry. Right now we only get the confirmation on the main a content. We could set an opacity to 0.5? or something?

juliushaertl commented:
@skjnldsv Looks really nice. One small issue I found while testing is that collapsed menus are only working when you click on the text. I think they should also collapse/expand when clicking the arrow symbol:

![2017-09-12-095915_236x128_scrot](https://user-images.githubusercontent.com/3404133/30314579-6d5a1c0e-97a1-11e7-89dd-f0215ffb05c6.png)

jancborchardt commented:
> One small issue I found while testing is that collapsed menus are only working when you click on the text. I think they should also collapse/expand when clicking the arrow symbol:

Every entry should be clickable on the whole height/width (extending to the left edge of the viewport, and to the right to the edge of the navigation, minus the action icons of course).

skjnldsv commented:
@juliushaertl It's because of my javascript! ^^
I made it as simple as I could, becuse i didn't wanted to bother too much.

@jancborchardt there isn't any js in core to handle the app-navigation doesn't it?

jancborchardt commented:
@skjnldsv not sure about the JS part but the clickable area is a pure CSS thing, no? Any idea ^ @BernhardPosselt @nextcloud/javascript?

skjnldsv commented:
The clickable area is yes. I need to fix a small think indeed so the link gets above the collapse button.
We could also get rid of that button entirely and add it css only, it could be usefull since it will simplify what the user needs to set-up for the menu! :)

skjnldsv committed:
Removed collapsible button requirements, and animate submenu deletion

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Animation fix

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Added some animation and deleted the collapse button.
We don't need it anymore, just use the class and everything is beeing taken care of! 🎉 

Some animation example:
![kazam_screencast_00001](https://user-images.githubusercontent.com/14975046/30332095-cbc0e0e2-97d9-11e7-859b-2a4196ffd9a5.gif)
![kazam_screencast_00002](https://user-images.githubusercontent.com/14975046/30332096-cbc7dda2-97d9-11e7-8c01-634a32dac1a8.gif)

skjnldsv committed:
Fixed comment

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Retro compatibility collapse button hide

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
@skjnldsv cool! Just one point about the opening animation for the collapsed menu: It seems like right now you only fade the entries in. But additionally there should be a short and quick slide animation, which moves the entries down. Otherwise entries below do a janky jump.

skjnldsv commented:
@jancborchardt That's a css issue. You can't animate height to auto, so you need to trick it with a max-height. BUT since you need to set a big height (like 2000px), the animation is slow to close and fast to appear. I used a cubic-bezier transition to speed up the close and slow the opening to trick the display a bit, but you can't really completely get rid of that! 😖 

We can reduce it a lot if we reduce the max-height, but it also limit the number of sub-items we can display. Should we define a limit?

Otherway, we can use js to animate the height.

skjnldsv committed:
Improve collapse animation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Improve the transition a bit by tricking the css. 
![kazam_screencast_00002](https://user-images.githubusercontent.com/14975046/30377098-e2cff150-988e-11e7-8c31-d6936744bdf6.gif)

skjnldsv committed:
no need for overflow, let the opacity trick the user and the animation smoothly push other items

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
Nice, that seems good! :) Yeah, it’s important we don’t set a max limit, because in the Mail app or News app you can potentially have unlimited entries in a folder.

skjnldsv commented:
> Yeah, it’s important we don’t set a max limit, because in the Mail app or News app you can potentially have unlimited entries in a folder.

If we want animation, it's either with a max-height or with js! :/
To be fair, actually the 2000px allows 45 subitems. Which is already an insanely big menu!

jancborchardt commented:
> To be fair, actually the 2000px allows 45 subitems. Which is already an insanely big menu!

Well … could be more ;) I’m sure @BernhardPosselt knows. :) I’d rather say we do it via JS if we can’t do it without limit, which would be severely confusing UX.

skjnldsv commented:
I don't know. Having it via js could create some issues with js injecting apps like contacts and mail. Since we add and update the dom constently, this is better handled by the app itselve.
I'm okay with the max limit to 2000, I can increase of course. 9999px could work the same!

jancborchardt commented:
@skjnldsv ok – then maybe 9999px? @nextcloud/designers does anyone of you have a better solution in mind?

skjnldsv committed:
Max height to 9999

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Added bullet class and fixed comments

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
![capture d ecran_2017-09-14_10-05-09](https://user-images.githubusercontent.com/14975046/30418710-35c101b0-9934-11e7-9a69-22ad5f0f3404.png)

Added the `app-navigation-entry-bullet` class !

skjnldsv committed:
Fix deletion/edit supperposition

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
So to sum it up
 - Two types of alignments
    - 1st level: 0
    - 2nd level: 56px
 - Icons and bullets
    - shift 1st levels by 44px
    - does NOT shift 2nd levels: icons & bullets are shifted to the left by 44px

![capture d ecran_2017-09-14_10-11-15](https://user-images.githubusercontent.com/14975046/30419859-247d4ac2-9938-11e7-908a-5a8d9a414f51.png)

skjnldsv committed:
Finally removed old legacy code

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@nextcloud/designers are you ok with this?
Currently, it's breaking a few apps, but there are simple fixes for 99% of them
- deck (absolute positioning of the menu to remove)
- tasks (editable form to redesign)
- mail (rebuild the whole menu since it wasn't ok on the first place)
- notes (works fine, but needs an update to fit the guidelines)
- contacts (works fine, but needs an update to fit the guidelines)
- calendar (works, but needs to fit the guidelines and a few design patches)

skjnldsv committed:
Hide bullet & icons if menu has loading icon

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Loading position fix

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix github color highlighter

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
It looks good except the indent for 2nd level. You changed that from the current state, right? Cause it looks just barely more than the 1st level, and mostly like a visual error. It should be indented more, like 88px.
Also, the whitespace right of the icon to the text is a bit much, or did you not change that?

skjnldsv commented:
@jancborchardt Hum, apparently I did!
The original padding is 32px.
Which brings us to a good question: Which one do you prefer?
![kazam_screenshot_00004](https://user-images.githubusercontent.com/14975046/30426746-77671532-994d-11e7-8134-afbf60bec4eb.png)![kazam_screenshot_00005](https://user-images.githubusercontent.com/14975046/30426744-77630c08-994d-11e7-9856-6dc8856895e7.png)![kazam_screenshot_00006](https://user-images.githubusercontent.com/14975046/30426745-776425b6-994d-11e7-8309-1ebd657a05d5.png)

skjnldsv commented:
I'm quite ok with #1, but we need to have a better indicator that this is a submenu since it's really hard to differentiate a 1st level with icon from a 2nd level with icon! 😕

jancborchardt commented:
Right, 32 is some middle ground because 88 is a bit much and leaves little space for the text.

However, it is important that we align the text of the entries of one level on the same vertical line – here the subentries:
![entries-aligned](https://user-images.githubusercontent.com/925062/30432337-a10ad2b6-9961-11e7-93d9-b18713ce6360.png)

And here with all entries. Essentially making .with-icon the default:
![entries-aligned-all](https://user-images.githubusercontent.com/925062/30432352-b1163948-9961-11e7-8f4c-6f1bad90af30.png)
It looks much cleaner and you clearly know which text is at which level. Also it pushes that ideally there are additional visual indicators like icons, favicons, etc.

skjnldsv commented:
I think it looks strange if no icons are here! :(
Is this what you want?
![capture d ecran_2017-09-14_15-49-06](https://user-images.githubusercontent.com/14975046/30433318-7677a6ac-9964-11e7-8ab3-275e2bf31c37.png)

jancborchardt commented:
@skjnldsv since most apps these days have icons, I would actually say that’s perfectly fine. It only looks a tiny bit strange if there are no icons at all there. And as said above »it pushes that ideally there are additional visual indicators like icons, favicons, etc.« :)

skjnldsv committed:
Icon shifting and unification of padding

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Okay then! Pushing in 3...2...1

skjnldsv committed:
Removed unused props

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Loader fix on main entry

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
Alright! This is pretty amazing! :) Found only small issues:
- [ ] First level »Deleted« text has no left whitespace, should line up with the normal entry text though
![screenshot from 2017-09-14 17-23-21](https://user-images.githubusercontent.com/925062/30438485-9fed7b12-9971-11e7-8b6a-6c7b9dc3df64.png)
- [ ] (This is just a detail about the template: We should encourage people actually saying what was deleted, like »Deleted foldername!« instead of just »Deleted!«. But if we do it right in the template, devs can also see we want that. :)
- [ ] Same left whitespace issue with the 1st level editable field:
![screenshot from 2017-09-14 17-29-42](https://user-images.githubusercontent.com/925062/30438770-5babdb50-9972-11e7-96f5-147beeacd237.png)
- [ ] The animation for editing is a bit strange, moving in from the right. Since it edits the text, I would rather fade because the text should be edited in place (basically making the input field appear around the text)
- [ ] Second level editable entry has obscured buttons (also, it says 1st level but is 2nd level :)
![screenshot from 2017-09-14 17-24-16](https://user-images.githubusercontent.com/925062/30438484-9fe2eddc-9971-11e7-9cc9-e1cc1117fb98.png)

Sidenote: Great animation with the collapsible folder icon moving away! :) Love it

And one general thought: Do we ever have an entry with sub-entries which is _not_ collapsible? As far as I can see, everywhere where we have sub-entries like in News and Mail they are all collapsible. And it would also be strange and inconsistent if that would not be possible. The only question is often if they are collapsed or expanded by default. But I would even think about removing the possibility to make them static, no?

skjnldsv committed:
Padding for deleted and editing entries

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
> Second level editable entry has obscured buttons (also, it says 1st level but is 2nd level :)

Can't reproduce, mine are correctly displayed! [Screenshot](https://user-images.githubusercontent.com/14975046/30443800-19b8776c-9981-11e7-8b3f-483e52ab31a6.png)

> Sidenote: Great animation with the collapsible folder icon moving away! :) Love it

Thanks! 😍 

>And one general thought: Do we ever have an entry with sub-entries which is not collapsible? As far as I can see, everywhere where we have sub-entries like in News and Mail they are all collapsible. And it would also be strange and inconsistent if that would not be possible. The only question is often if they are collapsed or expanded by default. But I would even think about removing the possibility to make them static, no?

The new behavior now is opened by default and close if the entry is collapsible.

> (This is just a detail about the template: We should encourage people actually saying what was deleted, like »Deleted foldername!« instead of just »Deleted!«. But if we do it right in the template, devs can also see we want that. :)

Won't fix it here, the template has been created just as a test. But we definitely needs to update the guidelines nextcloud/documentation/issues/333

skjnldsv committed:
Align edit input with its corresponding link text and fix animation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Fixed position and animation of the editing entry!
![kazam_screencast_00002](https://user-images.githubusercontent.com/14975046/30444387-164ba2dc-9983-11e7-96a4-709b09fc4b82.gif)

skjnldsv commented:
>  The hover on the confirm/cancel buttons needs the border-left too:

If so, we'll get double borders, I can try to trick the css a bit, but it's clearly a pain in the ![donkey](https://user-images.githubusercontent.com/14975046/30444659-fb065ba6-9983-11e7-90a0-207f32fa2695.png)

skjnldsv committed:
Fix entry edit buttons borders

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

MorrisJobke commented:
I pushed some fixes for the trashbin entry. We don't need the dirty hacks anymore now 🙌

MorrisJobke commented:
Could you have a look at the hacks in trash.css in the files_trashbin app?

![bildschirmfoto 2017-09-15 um 17 12 19](https://user-images.githubusercontent.com/245432/30490076-3c5039a0-9a39-11e7-947a-4de2941ae492.png)

skjnldsv commented:
@MorrisJobke If we don't have more items to add on the main list on the left, it could be great to use flex.
Like this we get them on the bottom for big height screens, but they turn as casual menu entry on small height.
WHich is usefull because fixed, they constrain the visibility of the menu which is annoying on small screens :)

![kazam_screencast_00002](https://user-images.githubusercontent.com/14975046/30508978-522ee6e8-9aa4-11e7-94d3-b5127e78586f.gif)

skjnldsv commented:
And I could add that we could provide a class to add which will automatically put your entry at the bottom. Useful for other apps! 😊

skjnldsv committed:
Flex to menu items

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
Nice! We could then also add back the quota icon I wanted to add in the PR of adding the quota bar? ;) cause without icon it now looks strange.

skjnldsv commented:
@jancborchardt I will definitely do that soon :)

skjnldsv commented:
Okay, There is faaaar more characters that break things.
Like if you put a smiley, this gets inserted in the db, but you can't delete it after. ⚠️ 
Or you're allowed to create an ab with an hashtag in it, but this break the contact creation in it once in a while...

We could use a regex that matches all alphabet characters including Japanese, Chinese, Norwegian, Turkish... but this require a special libs as javascript regex does NOT support the magic selector: `\p{L}`
We could also only allow European char, but this seems like a drastic change, and I'm not quite fond of it.

@nextcloud/contacts, what do you think?

skjnldsv committed:
Allow custom classes on nav entries

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Add pinned class to trashbin

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Add possibility to pin nav entries at the bottom

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Removed unwanted css and add quota icon

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Unwanted css for trashbin

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Unwanted css for trashbin

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Quota order

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix quota order

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
hack first class occurence .pinned

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Quota.svg

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@jancborchardt @MorrisJobke 
![capture d ecran_2017-09-17_17-24-07](https://user-images.githubusercontent.com/14975046/30522227-0df0e146-9bcd-11e7-9de5-aad6a8563988.png)

skjnldsv committed:
create special class forst first pinned class occurrence

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Okay we're good! :)
@nextcloud/designers please review!

jancborchardt commented:
@skjnldsv can you use this quota icon instead? https://github.com/nextcloud/server/blob/3f549609b9cca603418ad7b55401658627e99977/apps/files/img/quota.svg :) sorry I didn’t comment with that before.

skjnldsv commented:
@jancborchardt ho, I didn't saw this one. I prefer mine to be honest, I feel like the link you suggested isn't clear enough. 😕
It's just my opinion, whatever we decide is fine! 😉

skjnldsv commented:
Showed this to some people around me, and the icon you provided apparently looks like a clock! 😂

jancborchardt commented:
Yours is a pizza and too dark overall. ;) I mean, we can discuss long about it, but the point of mine is also that it could visualize the actual use quite clearly.

skjnldsv commented:
@jancborchardt but it's a 4cheese pizza!!! 😂
You mean dynamically? Like change the empty space depending on the current percentage?

MorrisJobke commented:
> You mean dynamically? Like change the empty space depending on the current percentage?

Let's do the quota icon adjustments in a separate PR. This one is already open long enough.

From my side it works nicely. Maybe in the future we can also add the settings entry to be on the bottom of the list and not fixed all time as proposed by @skjnldsv but for this PR it's fine. I tested it in IE11, Edge, Chrome, Safari and Firefox and it's awesome 👍

MorrisJobke commented:
@ChristophWurst @juliushaertl @nextcloud/designers Let's get this reviewed \o/

MorrisJobke commented:
Let me fix the failing unit tests.

MorrisJobke committed:
Fix unit tests

Signed-off-by: Morris Jobke <hey@morrisjobke.de>

MorrisJobke commented:
> Let me fix the failing unit tests.

Should work now.

skjnldsv commented:
Restarted the build.

skjnldsv commented:
Test successful, needs reviews! 🆘

jancborchardt commented:
> Let's do the quota icon adjustments in a separate PR.

I agree, the dynamic sizing was merely a remark.

@skjnldsv could you change the quota icon please? :)

nickvergessen commented:
In the spreed app the delete and copy
 button of the share link now is on the wrong row:

### Before
> ![bildschirmfoto vom 2017-09-19 10-52-25](https://user-images.githubusercontent.com/213943/30583984-b0e48cac-9d28-11e7-9a54-c93b46d452c8.png)

### After
> ![bildschirmfoto vom 2017-09-19 10-50-32](https://user-images.githubusercontent.com/213943/30583893-7c49414a-9d28-11e7-9be3-a26313cad638.png)

skjnldsv committed:
Quota icon by @jancborchardt

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@nickvergessen this should be fixed in the spreed app.

https://github.com/nextcloud/spreed/blob/master/css/style.css#L517-L520

Needs `position: relative;`

skjnldsv commented:
@jancborchardt Icon fixed! 😉

jancborchardt commented:
> this should be fixed in the spreed app

@skjnldsv I also thought that initially. But wouldn’t it make sense to set position: relative for the li by default?

Thanks for the icon change! :)

skjnldsv commented:
@jancborchardt I don't see an inconvenient to that.

skjnldsv committed:
Relative positionning to li

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
Actually one big issue: It doesn’t seem to be navigable via keyboard, or at least the :focus feedback is missing. Maybe something to do in a follow-up PR, but we should make sure the structure will allow for it.

Besides just two issues I could find:

![screenshot from 2017-09-19 15-45-44](https://user-images.githubusercontent.com/925062/30595523-be3f7bea-9d51-11e7-9f41-750ca26b83e1.png)
- [ ] The editable entry buttons are too narrow
- [ ] The edit icon has no hover effect


> I don't see an inconvenient to that.

What do you mean by that? :) I mean sure, every app can do it themselves, but if that inline link display is a pattern we want to use more, setting the position: relative; by default does not have a drawback, right?

skjnldsv commented:
I don't see an inconvenient! :p
Means I'm ok with this! ^^

Will work on the edit field as soon as I get back from work! :)

skjnldsv committed:
Fix editing field animation

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
> Besides just two issues I could find:
> ![screenshot from 2017-09-19 15-45-44](https://user-images.githubusercontent.com/925062/30595523-be3f7bea-9d51-11e7-9f41-750ca26b83e1.png)

DAMN you firefox!!!

skjnldsv committed:
damn you firefox default values

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@jancborchardt 
>  The edit icon has no hover effect

Well, it has the same effects as the three-dots menu? What would you suggest we add? :)

jancborchardt commented:
@skjnldsv huh, I can't see the hover effect there on the rename pencil icon. 3-dots-menu has slight transparency to full opacity. Will need to check again.

skjnldsv committed:
Avoid error undefined index classes in log

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@jancborchardt So? Where we at here? Could be great if we could merge !

jancborchardt reviewed (approved):
Looks all good now! :) Magnificent work!

jancborchardt commented:
What about the Drone failures @skjnldsv @MorrisJobke?

skjnldsv commented:
@jancborchardt not related to this pr I think.
I restarted it.

rullzer commented:
This has failing tests. let me have a look why

rullzer committed:
Fix tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

skjnldsv commented:
Thanks @rullzer , I'm an idiot! ^^'

rullzer commented:
@skjnldsv don't worry about it. we've all been there :)

skjnldsv commented:
🎉 
Thank you everyone! Next step: cleanup and documentation!! 🚋🚋🚋

pixelipo commented on apps/files/img/quota.svg:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Great work with this PR, @skjnldsv ! Just a small note - SVG should be optimized with Scour and it's missing `viewbox="0 0 16 16"` tag.

skjnldsv replied:
I used the scour command provided by the nextcloud optimise bash script :p

pixelipo replied:
Hmmm, that's strange, because it should remove the `<xml>` element as well as linebreaks....

skjnldsv replied:
scour --create-groups --enable-id-stripping --enable-comment-stripping --shorten-ids --remove-metadata apps/files/img/quota.svg > apps/files/img/quota-opt.svg

pixelipo replied:
That's old, @skjnldsv - you missed #5423 :wink:

skjnldsv replied:
Damn it!!! 😅

pixelipo commented on apps/files_trashbin/css/trash.css:
```diff
@@ -20,17 +20,3 @@
 	display: none;
 }
 
-/* move Deleted Files to bottom of sidebar */
```
only 3 declarations left in this file - can we move them somewhere else?

skjnldsv commented:
@MorrisJobke Should we backport?
I think we should. We can't have apps supporting only 13+, 12 should be good at least.

MorrisJobke commented:
As it is a feature I would say no - and for my opinion it is too much changed code for a back port

skjnldsv commented:
Damn, I should have done that earlier.
Should I publish a simple patch to fit the old guidelines and fix the potentials flaws of a migrated app? For retro compatibility?

MorrisJobke commented:
May be better - we only need to be careful about potential breakages of apps in the existing versions

jancborchardt commented:
Yeah, this can absolutely not be backported unfortunately. :\ But yes, a drop-in CSS file for apps while they need to support older versions still might be best.

skjnldsv commented:
Okay, I will provide a patch after I finished upgrading the other apps. When is 13 due?

MorrisJobke commented:
> When is 13 due?

~ 1 month feature freeze
~ 2 months release

skjnldsv commented:
Meep meep! 📯 
Time to speed up @skjnldsv ^^
